src/main/scala/io/iohk/ethereum/network/discovery/codecs/RLPCodecs.scala

```diff
@@ -49,53 +48,73 @@ object RLPCodecs {
       },
       {
         case list @ RLPList(items @ _*) if items.length >= 4 =>
-          val address = RLPDecoder.decode[Node.Address](list)
-          val id = RLPDecoder.decode[PublicKey](items(3))
+          val address = list.decodeAs[Node.Address]("address")
+          val id = items(3).decodeAs[PublicKey]("id")
```

Comment:
It's odd that we are obtaining the address from the whole list while id is one of it's items :thinking:

Reply:
It's because every RLP decoder ignores the "rest" of the fields, so Address takes as many fields as it wants (the first 3), but we know that after that comes the ID, a single value.

Reply:
Although it looks like in `go-ethereum` this is slightly different, with only the payloads such as `Neighbor` having a [Rest](https://github.com/ethereum/go-ethereum/blob/master/p2p/discover/v4wire/v4wire.go#L82) field, while the [Node](https://github.com/ethereum/go-ethereum/blob/master/p2p/discover/v4wire/v4wire.go#L134) itself doesn't. I followed our own [practice](https://github.com/input-output-hk/mantis/blob/develop/src/main/scala/io/iohk/ethereum/network/discovery/message.scala#L33) of always erring on the side of tolerating extra fields. We could fine tune the [DerivationPolicy](https://github.com/input-output-hk/mantis/blob/develop/src/main/scala/io/iohk/ethereum/rlp/RLPImplicitDerivations.scala#L11) on a case by case basis but I don't think it's worth the trouble.

Reply:
>I followed our own practice of always erring on the side of tolerating extra fields

Oh, I think we only followed that practice on the discovery messages, I haven't seen it on the TCP messages

If it's not something required by the protocol it's probably less error prone to not tolerate extra fields, but I wouldn't consider it a blocker for this PR and something that we can improve on later

Reply:
Ah, cool, didn't notice. Changed it now so by default the generated codecs reject extra list items and only the payloads accept the "rest" style fields.